src/Common/RWLock.cpp

```diff
@@ -278,42 +302,78 @@ void RWLockImpl::dropOwnerGroupAndPassOwnership(GroupsContainer::iterator group_
         readers_queue.erase(group_it);
         /// Prepare next phase
         if (!writers_queue.empty())
-        {
             wrlock_owner = writers_queue.begin();
-        }
         else
```

Comment:
Is this even possible anymore? If we removed all readers there must be a writer or nothing in the queue, but never another reader group.